# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Hrushi56/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

# 1. My Lane and Why

I selected the **Content Refresh** lane.

Many websites contain thousands of content pages, making it difficult for content teams to manually decide which pages should be updated first. The dataset contains information about content age, search visibility, traffic trends, engagement metrics, and page freshness. These features provide useful signals for identifying pages that are likely to benefit from a content refresh.

My goal is to build a system that helps prioritize which pages should be reviewed first. Rather than replacing human judgment, the project will support content teams by ranking pages that are most likely to need attention based on historical performance data.

# 2. The Research Question

## Research Question

Which content pages should be prioritized for refresh in order to maximize future search visibility and user engagement?

## Unit of Analysis

Each individual content page.

## Output

A priority score or ranking indicating which pages should be refreshed first.

## Decision

Decide which pages should receive a content update before others.

## Action

The content team reviews and updates the highest-priority pages first instead of manually checking thousands of pages.

## Cost of a Wrong Recommendation

If the model incorrectly recommends pages, the team may spend time updating pages that produce little improvement while high-impact pages continue losing traffic and search visibility. This results in wasted effort and missed opportunities.

## Why Machine Learning Can Help

The dataset contains thousands of pages and dozens of performance-related features. Identifying useful patterns manually is difficult because many variables interact with one another. Machine learning can learn these patterns from historical data and provide consistent recommendations to support better decision making.

In [3]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Dataset Shape:", df.shape)

print("\nTotal Content Pages:", len(df))

print("\nTrend Direction")
print(df["trend_direction"].value_counts())

print("\nFreshness Tier")
print(df["freshness_tier"].value_counts())

print("\nAverage CTR:", round(df["ctr"].mean(),2))
print("Average Position:", round(df["avg_position"].mean(),2))

Dataset Shape: (30000, 44)

Total Content Pages: 30000

Trend Direction
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Freshness Tier
freshness_tier
0-30      20480
91-180     9171
31-90       175
181+        174
Name: count, dtype: int64

Average CTR: 0.51
Average Position: 16.34


# 3. Quick Look at the Data

The dataset contains **30,000 content pages** with **44 features** describing search performance, engagement, freshness, and content characteristics.

Some observations from the dataset:

- Total pages: **30,000**
- Pages with a **downward trend:** **16,262**
- Average CTR: **0.51**
- Average search position: **16.34**
- Most pages belong to the **0–30 day freshness tier**, indicating many pages are relatively recent.

These statistics suggest that there are enough examples of declining content to make the Content Refresh lane a meaningful machine learning problem.

# 4. Careful Words

This project aims to recommend which pages should be reviewed for a content refresh based on historical performance data.

The model will not guarantee improvements in search rankings or traffic because many external factors influence search performance, including search engine algorithm changes, competitor activity, seasonality, and future user behavior.

The recommendations produced by the model should be treated as decision support rather than automatic decisions. Human review remains important before updating content.

# 5. Self Check

✅ Selected one of the predefined project lanes.

✅ Clearly defined the research question.

✅ Specified the unit of analysis.

✅ Explained the decision and resulting action.

✅ Described the cost of an incorrect recommendation.

✅ Included real statistics from the dataset.

✅ Explained why machine learning is appropriate.

✅ Used careful language about the limitations of the project.